### PR TITLE
replaced `skip_serializing_if` with `skip_serializing_none`

### PR DESCRIPTION
**Pull request by Seba-T**
all the single `#[serde(skip_serializing_if = "Option::is_none")]` got replaced with the more readable 
`#[skip_serializing_none]`, which has to be specified only once per struct 

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/core/Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ tracing-subscriber = { workspace = true, features = ["time", "json"] }
 urlencoding = { workspace = true }
 kanidm_utils_users = { workspace = true }
 uuid = { workspace = true, features = ["serde", "v4" ] }
+serde_with = "3.1.0"
```
Comment (yaleman) [inline server/core/Cargo.toml:52]: This should be a workspace import

File: server/lib/Cargo.toml [hunk 2]
```diff
@@ -65,6 +65,7 @@ uuid = { workspace = true, features = ["serde", "v4" ] }
 webauthn-rs = { workspace = true, features = ["resident-key-support", "preview-features", "danger-credential-internals"] }
 webauthn-rs-core = { workspace = true }
 zxcvbn = { workspace = true }
+serde_with = "3.1.0"
```
Comment (yaleman) [inline server/lib/Cargo.toml:68]:
```suggestion
serde_with = { workspace = true }
```

**Comment by Seba-T** [comment]
Alright I think I fixed them all!

**Review by yaleman — APPROVED** [review]
Sweet! Less code, more ergonomic, more better.

**Pull request merged**